apps/emqx_redis/src/emqx_redis.erl

```diff
@@ -242,24 +243,57 @@ on_get_status(_InstId, #{type := cluster, pool_name := PoolName}) ->
                 [] ->
                     disconnected;
                 [_ | _] ->
-                    Health = eredis_cluster:ping_all(PoolName),
-                    status_result(Health)
+                    do_cluster_status_check(PoolName, State)
             end;
         false ->
             disconnected
     end;
-on_get_status(_InstId, #{pool_name := PoolName}) ->
-    Health = emqx_resource_pool:health_check_workers(PoolName, fun ?MODULE:do_get_status/1),
-    status_result(Health).
+on_get_status(_InstId, #{pool_name := PoolName} = State) ->
+    Timeout = 1000,
+    Workers = [Worker || {_WorkerName, Worker} <- ecpool:workers(PoolName)],
+    DoPerWorker =
+        fun(Worker) ->
+            case ecpool_worker:client(Worker) of
+                {ok, Conn} ->
+                    erlang:is_process_alive(Conn) andalso
+                        ecpool_worker:exec(Worker, fun ?MODULE:do_get_status/1, Timeout);
+                Error ->
+                    Error
+            end
+        end,
+    {ok, Results} =
+        try
+            {ok, emqx_utils:pmap(DoPerWorker, Workers, Timeout)}
+        catch
+            exit:timeout ->
+                {error, timeout}
+        end,
```

Comment:
Nit: this seems to be what [`emqx_resource_pool:health_check_workers/4`](https://github.com/emqx/emqx/blob/f969a4ef5ea79d4e79a1cbbd131b709091b80cf7/apps/emqx_resource/src/emqx_resource_pool.erl#L79-L101) does with `return_values => true`.

Reply:
Ok, I managed to miss that. I will try to simplify.

Reply:
Fixed